packages/@aws-cdk/cx-api/lib/private/flag-modeling.ts

```diff
@@ -0,0 +1,39 @@
+export enum FlagType {
+  /**
+   * Change the default behavior of the API
+   *
+   * The old behavior is not disrecommended, and possible to achieve with source
+   * code changes. Also valid for changes that don't affect CloudFormation, but
+   * the CXAPI contract.
+   */
+  ApiDefault,
+
+  /**
+   * Address a bug/introduce a recommended change
+   *
+   * The old behavior is no longer recommended. The only way to achieve it is by
+   * keeping the flag at the legacy value.
+   */
+  BugFix,
+
+  /**
+   * Advertise the presence of this context option in `cdk.json`
+   */
+  VisibleContext,
+};
+
+export interface FlagInfo {
```

Comment:
I know we are not even thinking about `v3`, but should this be more flexible with versions?

Reply:
It's not really possible in a way that can be properly typechecked with TypeScript.

Even if we did, adding a `v3?: string` to the existing structures would be trivial.